repositories.bzl

```diff
@@ -1,22 +1,295 @@
+
+load("@io_bazel_rules_go//go:def.bzl", "go_prefix")
+go_prefix("github.com/googleapis/googleapis")
+
+load("@org_pubref_rules_protobuf//gogo:rules.bzl", "gogoslick_proto_library")
+
+gogoslick_proto_library(
+    name = "google/rpc",
+    protos = [
+        "google/rpc/code.proto",
+        "google/rpc/error_details.proto",
+        "google/rpc/status.proto",
+    ],
+    importmap = {
+        "google/protobuf/any.proto": "github.com/gogo/protobuf/types",
+        "google/protobuf/duration.proto": "github.com/gogo/protobuf/types",
+    },
+    imports = [
+        "../../external/com_github_google_protobuf/src",
+    ],
+    inputs = [
+        "@com_github_google_protobuf//:well_known_protos",
+    ],
+    deps = [
+        "@com_github_gogo_protobuf//types:go_default_library",
+    ],
+    verbose = 0,
+)
+
+load("@org_pubref_rules_protobuf//cpp:rules.bzl", "cc_proto_library")
+
+cc_proto_library(
+    name = "cc_status_proto",
+    protos = [
+        "google/rpc/status.proto",
+    ],
+    imports = [
+        "../../external/com_github_google_protobuf/src",
+    ],
+    verbose = 0,
+)
+
+filegroup(
+    name = "status_proto",
+    srcs = [ "google/rpc/status.proto" ],
+)
+
+filegroup(
+    name = "code_proto",
+    srcs = [ "google/rpc/code.proto" ],
+)
+"""
+    native.new_git_repository(
+        name = "com_github_googleapis_googleapis",
+        build_file_content = GOOGLEAPIS_BUILD_FILE,
+        commit = "13ac2436c5e3d568bd0e938f6ed58b77a48aba15", # Oct 21, 2016 (only release pre-dates sha)
+        remote = "https://github.com/googleapis/googleapis.git",
+    )
+
+def go_x_tools_imports_repositories():
+    BUILD_FILE = """
+package(default_visibility = ["//visibility:public"])
+load("@io_bazel_rules_go//go:def.bzl", "go_binary")
+load("@io_bazel_rules_go//go:def.bzl", "go_prefix")
+
+go_prefix("golang.org/x/tools")
+
+licenses(["notice"])  # New BSD
+
+exports_files(["LICENSE"])
+
+go_binary(
+    name = "goimports",
+    srcs = [
+        "cmd/goimports/doc.go",
+        "cmd/goimports/goimports.go",
+        "cmd/goimports/goimports_gc.go",
+        "cmd/goimports/goimports_not_gc.go",
+    ],
+    deps = [
+        "@org_golang_x_tools//imports:go_default_library",
+    ],
+)
+"""
+    # bazel rule for fixing up cfg.pb.go relies on running goimports
+    # we import it here as a git repository to allow projection of a
+    # simple build rule that will build the binary for usage (and avoid
+    # the need to project a more complicated BUILD file over the entire
+    # tools repo.)
+    native.new_git_repository(
+	name = "org_golang_x_tools_imports",
+        build_file_content = BUILD_FILE,
+        commit = "e6cb469339aef5b7be0c89de730d5f3cc8e47e50",  # Jun 23, 2017 (no releases)
+        remote = "https://github.com/golang/tools.git",
+    )
+
+# This function should be used by others to use mock mixer.
+def mixer_test_repositories(use_local_api=False):
+    go_repositories()
```

Comment:
Dont't want to put too many things in WORKSPACE.  For example, istio/proxy WORKSAPCE are c++ related.